Microsoft.Toolkit.Uwp.UI.Controls/Carousel/Carousel.cs

```diff
@@ -294,7 +292,7 @@ public Carousel()
                 var itemsPanel = carouselControl.GetItemsPanel();
 
                 // Prioritize the SelectedItem over the SelectedIndex
-                if (SelectedItem != null && Items[carouselControl.SelectedIndex] != SelectedItem)
+                if (SelectedItem != null && (Items.Count() <= carouselControl.SelectedIndex || Items[carouselControl.SelectedIndex] != SelectedItem))
```

Comment:
This line is a bit weird. Should it be `Items.Count() > carouselControl.Selected.Index`, (Instead of the less and equal)? Maybe also check if SelectedIndex <0